### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by Turbo87**
This PR adds documentation comments to the `user` module.

bildschirmfoto 2015-11-14 um 00 48 58

### CODE REVIEW

**Comment by Gankro** [comment]
:confetti_ball: awesome!

File: src/bin/delete-crate.rs [hunk 1]
```diff
@@ -18,6 +18,7 @@ use std::io::prelude::*;
 
 use cargo_registry::Crate;
 
+#[allow(dead_code)]
 fn main() {
```
Comment (Gankro) [inline src/bin/delete-crate.rs]: iirc the "proper" way to do this is `#[cfg(not(test))]`. May be misremembering for `doc`.

File: src/user/mod.rs [hunk 2]
```diff
@@ -20,6 +20,8 @@ pub use self::middleware::{Middleware, RequestUser};
 
 pub mod middleware;
 
+/// The model representing a row in the `users` database table.
+///
```
Comment (Gankro) [inline src/user/mod.rs]: We don't add trailing ///'s

**Comment by Gankro** [comment]
No opinion on the middleware changes. (I also don't have review rights anyway, so y'all'll need to wait on someone else for that)

File: src/bin/delete-crate.rs (hunk 1 above)
Comment (Turbo87) [inline src/bin/delete-crate.rs]: if I replace this with `#[cfg(not(test))]` I'm getting the following soft of errors again:

```
src/bin/delete-version.rs:22:1: 31:2 error: function is never used: `main`, #[deny(dead_code)] on by default
```

File: src/user/mod.rs (hunk 2 above)
Comment (Turbo87) [inline src/user/mod.rs]: I felt like it was much easier to read the doc if they have a little room to breathe, but if this is not desired I can remove the extra line

**Comment by alexcrichton** [comment]
Thanks!

**Pull request merged**